rotkehlchen/chain/ethereum/modules/compound/compound.py

```diff
@@ -0,0 +1,234 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Optional
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.ethereum.modules.compound.constants import CPT_COMPOUND
+from rotkehlchen.chain.ethereum.modules.compound.utils import CompoundBalance
+from rotkehlchen.chain.ethereum.modules.compound.v2.compound import CompoundV2
+from rotkehlchen.chain.ethereum.modules.compound.v3.compound import CompoundV3
+from rotkehlchen.constants import ZERO
+from rotkehlchen.db.filtering import EvmEventFilterQuery
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.history.events.structures.evm_event import EvmEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType
+from rotkehlchen.history.price import query_usd_price_zero_if_error
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, Timestamp
+from rotkehlchen.utils.interfaces import EthereumModule
+from rotkehlchen.utils.misc import ts_ms_to_sec
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.defi.structures import GIVEN_DEFI_BALANCES, GIVEN_ETH_BALANCES
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.premium.premium import Premium
+    from rotkehlchen.user_messages import MessagesAggregator
+
+ADDRESS_TO_ASSETS = dict[ChecksumEvmAddress, dict[Asset, Balance]]
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class Compound(EthereumModule, CompoundV2, CompoundV3):
+    """Compound integration module
+
+    https://compound.finance/docs#guides
+    """
+
+    def __init__(
+            self,
+            ethereum_inquirer: 'EthereumInquirer',
+            database: 'DBHandler',
+            premium: Optional['Premium'],
+            msg_aggregator: 'MessagesAggregator',
+    ):
+        CompoundV2.__init__(self, ethereum_inquirer=ethereum_inquirer)
```

Comment:
you are combining two modules into one here. But I see we do similar with aave .... hmm we should probably change this at some point and allow drilling down into different versions.

Though this may make it harder for users so not sure. Will let it be for now.

Reply:
Yeah. Aave have a common file for both versions